Userland/Libraries/LibCore/CMakeLists.txt

```diff
@@ -62,7 +62,7 @@ else()
 endif()
 
 serenity_lib(LibCore core)
-target_link_libraries(LibCore PRIVATE LibCrypt LibSystem)
+target_link_libraries(LibCore PRIVATE LibCrypt LibSystem LibURL)
```

Comment:
This dependency isn't ideal IMO - other than the idea of LibCore depending on as little as possible, it will cause problems in the future (although, I suppose this MR isn't really changing the dependency graph anyway)

LibUnicode depends on the code generators, which depend on LibCore. And LibURL _wants_ to depend on LibUnicode. And that would leave us in a circular dependency. So yeah, I think will still need some future work here, although I'm not 100% sure on what to do about it yet.

Reply:
Yeah that circular dependency between generators and LibCore has gotten increasingly annoying. We can't use TZDB data in `Core::DateTime` for the same reason. Some ideas that have been floating in my head:

1. Rewrite the generators in python
2. Maybe we can get by with a "minimal" LibCore for code generators, i.e. link in streams, `ArgsParser,` etc. - just the things they use.

Reply:
Okay, yeah I think I might report a new issue for that to investigate / think about later. Getting that type of thing working was the main motivation behind working on this, but it seems we're not quite there just yet :^)

Reply:
I've raised an issue for it here: https://github.com/SerenityOS/serenity/issues/23625